pyspecdata/lmfitdata.py

```diff
@@ -303,18 +280,31 @@ def fit(self):
         # }}}
         return self
 
-    def run_lambda(self, pars):
-        """actually run the lambda function we separate this in case we want
-        our function to involve something else, as well (e.g. taking a Fourier
-        transform)"""
-        logging.debug(strm(self.getaxis(j) for j in self.variable_names))
-        return self.fitfunc_multiarg_v2(
-            *(self.getaxis(j) for j in self.variable_names), **pars.valuesdict()
-        )
+    def run_lambda(self, pars, thistaxis = None):
+        """actually run the lambda function that calculates the model data.
+        Note that the name of the variable along which the model data is calculated
+        (as opposed to "parameter" is set by variable_names parameter).
+
+        .. note::
+            we had separates this in case we want
+            our function to involve something else, as well (e.g. taking a Fourier
+            transform).  Unknown if there are still two steps in this way.
+
+        """
+        if len(self.variable_names) > 1:
+            print("We are only allowing for one variable for now")
+        elif thistaxis is None:
+            return self.fitfunc_multiarg_v2(
+                    *(self.getaxis(j) for j in self.variable_names), **pars.valuesdict())
+        else:    
+            logging.debug(strm(self.getaxis(j) for j in self.variable_names))
+            return self.fitfunc_multiarg_v2(
+                    thistaxis, **pars.valuesdict()
+            )
 
     def residual(self, pars, x, y, sigma=None):
         "calculate the residual OR if data is None, return fake data"
-        fit = self.run_lambda(pars)
+        fit = self.run_lambda(pars,thistaxis = None)
```

Comment:
you should be able to just leave this line alone (even before, this was your default, so there should be no need to change) --> this will not work with the change I suggest above.

Reply:
I think you're saying "no need to change" and when I leave this both examples work

Reply:
no, with suggestions below, your change will not work.  Leave the way it was before the PR (red line)

Reply:
since this was a tangential issue, I just did it in my new commit.